src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -12,12 +12,18 @@
 'use strict';
 
 var ReactOwner = require('ReactOwner');
+var warning = require('warning');
 
 var ReactRef = {};
 
 function attachRef(ref, component, owner) {
   if (typeof ref === 'function') {
-    ref(component.getPublicInstance());
+    var instance = component.getPublicInstance();
+    warning(
+      instance !== null,
+      'Stateless function components cannot have refs.'
```

Comment:
But I do think I should probably include the component/ref name just like https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L1187-L1190